### PR TITLE
moved the .h and .cpp files from the Pathfinding and src sub-folders …

### LINKED ISSUE #2707: Update ENLSVG build configuration
### Description of the task

ENLSVG is the path planner we use to navigate around non-moving obstacles (like the defence area). We got this algorithm from an external repo and it lives in our `Software/src/extlibs/enlsvg` folder. As such, it does not follow our build file configuration that we maintain in the rest of the codebase. Currently the header files live in `Software/src/extlibs/enlsvg/Pathfinding` while the source files live in `Software/src/extlibs/enlsvg/src`. We don't follow this pattern anywhere else in our codebase and we should update the build configuration in preparation for making our own version of ENLSVG.

This ticket involves moving the .h files in the `Pathfinding` folder and .cpp files in the `src` folder into `Software/src/extlibs/enlsvg/`. Then update `Software/src/extlibs/enlsvg/BUILD` to find these files in their new locations.

### Acceptance criteria

- [x] Move .h files from `Software/src/extlibs/enlsvg/Pathfinding` into `Software/src/extlibs/enlsvg`
- [x] Move .cpp files from `Software/src/extlibs/enlsvg/src` into `Software/src/extlibs/enlsvg`
- [x] Update `Software/src/extlibs/enlsvg/BUILD` to find these new files

### PR DESCRIPTION
**Pull request by ericlim0329**
…into the root directory.

After, went into the BUILD file to change some code to find the new locations of the files finally, deleted the Pathfinding and src subfolders or subdirectories from extlib/enlsvg

### Testing Done

resolves #2707 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by nimazareian — COMMENTED** [review]
Overall looks good to me! Just seems like you have a merge conflict with master branch. You will need to `git pull upstream master` and solve the merge conflicts and push the fix so your PR can be merged. Also, don't forget to tag your ticket in the PR description (adding `resolves ` 
 keyword before it).

File: src/extlibs/enlsvg/ENLSVGAlgorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+=======
+#include "extlibs/enlsvg/Pathfinding/enlsvg_algorithm.h"
+
+#include <algorithm>
+
+#include "extlibs/enlsvg/Pathfinding/enlsvg_graph.h"
+#include "extlibs/enlsvg/Pathfinding/grid.h"
+#include "extlibs/enlsvg/Pathfinding/indirect_heap.h"
+#include "extlibs/enlsvg/Pathfinding/line_of_sight_scanner.h"
+>>>>>>> 2330f76f64c53b1a23c58fb60b72d47482ba3da5:src/extlibs/enlsvg/enlsvg_algorithm.cpp
```
Comment (itsarune) [inline src/extlibs/enlsvg/ENLSVGAlgorithm.cpp]: you have git conflict markers, which means that you have some conflicts that haven't been resolved yet

File: src/extlibs/enlsvg/ENLSVGAlgorithm.h [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+#include "extlibs/enlsvg/IndirectHeap.h"
+#include "extlibs/enlsvg/LineOfSightScanner.h"
+#include "extlibs/enlsvg/PathfindingDataTypes.h"
+=======
+#include "enlsvg_graph.h"
+#include "grid.h"
+#include "indirect_heap.h"
+#include "line_of_sight_scanner.h"
+#include "pathfinding_data_types.h"
+>>>>>>> 2330f76f64c53b1a23c58fb60b72d47482ba3da5:src/extlibs/enlsvg/enlsvg_algorithm.h
```
Comment (itsarune) [inline src/extlibs/enlsvg/ENLSVGAlgorithm.h]: git conflict markers here

File: src/extlibs/enlsvg/enlsvg_algorithm.h [hunk 3]
```diff
@@ -5,11 +5,19 @@
 #include <limits>
 #include <vector>
 
+<<<<<<< HEAD:src/extlibs/enlsvg/ENLSVGAlgorithm.h
```
Comment (itsarune) [inline src/extlibs/enlsvg/enlsvg_algorithm.h]: you've got a conflict here, basically an earlier PR renamed these files, so git's a bit confused on which set to use

File: src/extlibs/enlsvg/LineOfSightScanner.h [hunk 4]
```diff
@@ -0,0 +1,132 @@
+#ifndef _LINE_OF_SIGHT_SCANNER_H_
```
Comment (itsarune) [inline src/extlibs/enlsvg/LineOfSightScanner.h]: this file is a duplicate of `line_of_sight_scanner.h` and we prefer the line_of_sight_scanner naming

File: src/extlibs/enlsvg/LineOfSightScanner.cpp [hunk 5]
```diff
@@ -0,0 +1,556 @@
+#include "extlibs/enlsvg/LineOfSightScanner.h"
```
Comment (itsarune) [inline src/extlibs/enlsvg/LineOfSightScanner.cpp]: this file is overwritten with "line_of_sight_scanner.cpp", so it shouldn't exist

File: src/extlibs/enlsvg/IndirectHeap.cpp [hunk 6]
```diff
@@ -0,0 +1,107 @@
+#include "extlibs/enlsvg/IndirectHeap.h"
```
Comment (itsarune) [inline src/extlibs/enlsvg/IndirectHeap.cpp]: we have indirect_heap.cpp which is a successor to this file, so you can delete this file

File: src/extlibs/enlsvg/Grid.h [hunk 7]
```diff
@@ -0,0 +1,372 @@
+#ifndef _GRID_H_
```
Comment (itsarune) [inline src/extlibs/enlsvg/Grid.h]: we have grid.h which is a successor to this file, so we can delete this one

File: src/extlibs/enlsvg/Grid.cpp [hunk 8]
```diff
@@ -0,0 +1,25 @@
+#include "extlibs/enlsvg/Grid.h"
```
Comment (itsarune) [inline src/extlibs/enlsvg/Grid.cpp]: we have grid.cpp which is a duplicate of this file, so we can delete this one

File: src/extlibs/enlsvg/Fraction.cpp [hunk 9]
```diff
@@ -0,0 +1,96 @@
+#include "extlibs/enlsvg/Fraction.h"
```
Comment (itsarune) [inline src/extlibs/enlsvg/Fraction.cpp]: we have fraction.cpp so we can delete this file

File: src/extlibs/enlsvg/ENLSVGGraph.h [hunk 10]
```diff
@@ -0,0 +1,138 @@
+#ifndef _ENLSVG_GRAPH_H_
```
Comment (itsarune) [inline src/extlibs/enlsvg/ENLSVGGraph.h]: we have enlsvg_graph.h so we can delete this file

File: src/extlibs/enlsvg/ENLSVGGraph.cpp [hunk 11]
```diff
@@ -0,0 +1,358 @@
+#include "extlibs/enlsvg/ENLSVGGraph.h"
```
Comment (itsarune) [inline src/extlibs/enlsvg/ENLSVGGraph.cpp]: we have enlsvg_graph.cpp so we can delete this file

File: src/extlibs/enlsvg/ENLSVGAlgorithm.h [hunk 12]
```diff
@@ -0,0 +1,199 @@
+#ifndef _ENLSVG_ALGORITHM_H_
```
Comment (itsarune) [inline src/extlibs/enlsvg/ENLSVGAlgorithm.h]: we have enlsvg_algorithm.h, so we can delete this file

File: src/extlibs/enlsvg/ENLSVGAlgorithm.cpp [hunk 13]
```diff
@@ -0,0 +1,457 @@
+<<<<<<< HEAD:src/extlibs/enlsvg/ENLSVGAlgorithm.cpp
```
Comment (itsarune) [inline src/extlibs/enlsvg/ENLSVGAlgorithm.cpp]: we have enlsvg_algorith.cpp, so we can delete this file

File: src/extlibs/enlsvg/ENLSVG.h [hunk 14]
```diff
@@ -0,0 +1,8 @@
+#ifndef _ENLSVG_H_
```
Comment (itsarune) [inline src/extlibs/enlsvg/ENLSVG.h]: we have enlsvg.h, so we can delete this file

**Review by itsarune — COMMENTED** [review]
Seems you got caught in a very nasty git merge conflict which was caused by an earlier PR that went in that renamed file names according to our convention "lower_case.cpp/.h"

generally, try resolving it by deleting files that are duplicated that start with "PascalCase.cpp/.h" that have a "snake_case.cpp/.h" equivalent

**Review by nimazareian — APPROVED** [review]
LGTM! 💯

**Review by itsarune — APPROVED** [review]
nice, good job!

**Pull request merged**